contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -143,6 +155,308 @@ contract OETHVaultCore is VaultCore {
         // Burn OETH from user (including fees)
         oUSD.burn(msg.sender, _amount);
 
+        // Prevent insolvency
         _postRedeem(_amount);
     }
+
+    /**
+     * @notice Request an asynchronous withdrawal of WETH in exchange for OETH.
+     * The OETH is burned on request and the WETH is transferred to the withdrawer on claim.
+     * This request can be claimed once the withdrawal queue's `claimable` amount
+     * is greater than or equal this request's `queued` amount.
+     * There is no minimum time or block number before a request can be claimed. It just needs
+     * enough WETH liquidity in the Vault to satisfy all the outstanding requests to that point in the queue.
+     * OETH is converted to WETH at 1:1.
+     * @param _amount Amount of OETH to burn.
+     * @param requestId Unique ID for the withdrawal request
+     * @param queued Cumulative total of all WETH queued including already claimed requests.
+     */
+    function requestWithdrawal(uint256 _amount)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256 requestId, uint256 queued)
+    {
+        // Burn the user's OETH
+        // This also checks the requester has enough OETH to burn
+        oUSD.burn(msg.sender, _amount);
+
+        requestId = withdrawalQueueMetadata.nextWithdrawalIndex;
+        queued = withdrawalQueueMetadata.queued + _amount;
+
+        // store the next withdrawal request
+        withdrawalQueueMetadata.nextWithdrawalIndex = uint128(requestId + 1);
```

Comment:
These casts will silently clip values. While I think uint128's should be safe for these values, I think we should default to using some kind of safe cast that reverts on overflows.

Reply:
at a rate of one billion requests per second, it would still take 1.08 × 10^22 years to process 2^128  requests.

I've added a `SafeCast.toUint128` so we don't have to deal with this in bug bounties

Reply:
I'm thinking our casts in general, not just this one line. Basically unless we have a serious reason for not doing a safe cast, we should use a safe cast. Gas overhead is minimal, though I'm sad it makes the code uglier. Maybe a `using` would help with the readability?

Reply:
I prefer to be explicit with `SafeCast.toUint128(queued)`.
If you have a lot of using libraries it can be hard to know what is being used from which library.